lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -175,7 +170,7 @@ export class ResponseHandler {
         );
 
         const reqEnvironment = authority.canonicalAuthorityUrlComponents.HostNameAndPort;
-        const env = EnvironmentAliases.includes(reqEnvironment) ? PreferredCacheEnvironment : reqEnvironment;
+        const env = TrustedAuthority.getCloudDiscoveryMetadata(reqEnvironment) ? TrustedAuthority.getCloudDiscoveryMetadata(reqEnvironment).preferred_cache : reqEnvironment;
```

Comment:
Check this later - Ideally by this point you should have the `reqEnvironment` in the trusted list, else it should throw an error. As per what we discussed, say users sets `myauthority.com/<tenantid>` as the `authority` in `configuration`, we should have created the below entry:

```javascript
this.CloudDiscoveryMetadata["myauthority.com"] = {
    preferred_cache: "myauthority.com",
    preferred_network: "myauthority.com",
    aliases: ["myauthority.com",]
};
```

Reply:
Done. Throws error if it's not found.